client/blocks/privacy-policy-banner/index.jsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getPreference, isFetchingPreferences } from 'state/preferences/selectors';
+import { savePreference } from 'state/preferences/actions';
+import { identity } from 'lodash';
+import Banner from 'components/banner';
+import config from 'config';
+import PrivacyPolicyDialog from './privacy-policy-dialog';
+import QueryPrivacyPolicy from 'components/data/query-privacy-policy';
+import { getPrivacyPolicyByEntity } from 'state/selectors';
+import { AUTOMATTIC_ENTITY, PRIVACY_POLICY_PREFERENCE } from './constants';
+
+class PrivacyPolicyBanner extends Component {
+	static propTypes = {
+		isPolicyAlreadyAccepted: PropTypes.bool,
+		privacyPolicy: PropTypes.object,
+		privacyPolicyId: PropTypes.string,
+		text: PropTypes.string,
+		translate: PropTypes.func,
+	};
+
+	static defaultProps = {
+		privacyPolicy: {},
+		text: '',
+		translate: identity,
+	};
+
+	state = {
+		showDialog: false,
+	};
+
+	acceptUpdates = () => {
+		if ( ! this.props.privacyPolicyId ) {
+			return;
+		}
+
+		const { privacyPolicyId, privacyPolicyUserStatus } = this.props;
+		this.props.acceptPrivacyPolicy( privacyPolicyId, privacyPolicyUserStatus );
+	};
+
+	openPrivacyPolicyDialog = () => this.setState( { showDialog: true } );
+
+	closePrivacyPolicyDialog = () => {
+		this.setState( { showDialog: false } );
+		this.acceptUpdates();
+	};
+
+	render() {
+		if ( ! config.isEnabled( 'privacy-policy' ) ) {
+			return null;
+		}
+
+		const {
+			fetchingPreferences,
+			isPolicyAlreadyAccepted,
+			moment,
+			privacyPolicy,
+			translate,
+		} = this.props;
+
+		if ( fetchingPreferences ) {
+			return null;
+		}
+
+		// check if the user has already accepted/read the privacy policy.
+		if ( isPolicyAlreadyAccepted === true && ! config.isEnabled( 'privacy-policy/test' ) ) {
+			return <QueryPrivacyPolicy />;
```

Comment:
Doesn't the banner always needs a query on mount? What happens if they accept a policy, and then we push out another update?

Reply:
Well ... the period of a new privacy policy could take time. Let's say six months. And the period used to show the banner is one month approx, it's defined in the endpoint response body, and it could be different for each version though.
But regarding the implementation, I think you're right. Maybe we should query when the component is mounted.